projects/libpng-proto/png_proto_mutator.cc

```diff
@@ -0,0 +1,34 @@
+#include "libprotobuf-mutator/src/libfuzzer/libfuzzer_macro.h"
+#include "png_fuzz_proto.pb.h"
+
+struct PngProtoMutator
+{
+	PngProtoMutator(
+		google::protobuf::Descriptor const* _desc,
+		std::function<void(google::protobuf::Message*, unsigned int)> _callback
+	)
+	{
+		protobuf_mutator::libfuzzer::RegisterPostProcessor(_desc, _callback);
```

Comment:
Actually here I'd recommend to use 
PostProcessorRegistration like in https://github.com/google/libprotobuf-mutator/blob/master/examples/libfuzzer/libfuzzer_example.cc
protobuf_mutator::libfuzzer::RegisterPostProcessor is kind of internal details, I need to rename it according 
PostProcessorRegistration should be a public interface, also it will cast message for you

Reply:
Ah I guess the problem with this approach is that I need to define the mutator callbacks in the fuzzing harness source file.

I would like to separate the mutators in a separate file and retain the fuzzing harness as-is, so that I can create a new fuzzing binary by simply linking the source file containing custom mutators. WDYT?